app/views/api_views.py

```
@@ -20,6 +20,34 @@ def autocomplete_name(request):
         return HttpResponse('', content_type="application/json")
 
 
+@login_required(login_url='/login')
+@require_GET
+def donor_names(request):
+    try:
+        donor_names = [donor.donor_name for donor in Donor.objects.all()]
+        return JsonResponse({'donorNames': donor_names},
+                            content_type="application/json")
+    except:
+        return HttpResponse('', content_type="application/json")
+
+
+@login_required(login_url='/login')
+@require_GET
+def device_names(request):
+    try:
+        device_objs = []
+        for device in ItemDevice.objects.all():
+            if device.dtype is not None:
+                deviceInfo = device.dtype.device_type + \
+                             " (" + device.make + "-" + device.model + ")"
+                device_objs.append(deviceInfo)
+
+        return JsonResponse({'deviceNames': device_objs},
+                            content_type="application/json")
+    except Exception as e:
+        return HttpResponse('', content_type="application/json")
```

Please either throw an error response or return empty list

---

Also please return as JsonResponse or HttpResponse with error